Make the <marquee> element spec-compliant by adding overflow: hidden

AlexsanderDamaceno opened the pull request:
#### 25d80dcb4ad37d0780f9a88f05c01de0a1935c04
<pre>
Make the &lt;marquee&gt; element spec-compliant by adding overflow: hidden
<a href="https://bugs.webkit.org/show_bug.cgi?id=283811">https://bugs.webkit.org/show_bug.cgi?id=283811</a>

Reviewed by Tim Nguyen.

Make the &lt;marquee&gt; element spec-compliant by adding  missing overflow: hidden property.

&lt;marquee&gt; element spec: <a href="https://html.spec.whatwg.org/#the-marquee-element-2">https://html.spec.whatwg.org/#the-marquee-element-2</a>

* Source/WebCore/css/html.css:
(marquee):
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjust const):

Canonical link: <a href="https://commits.webkit.org/293513@main">https://commits.webkit.org/293513@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/86bffb52671f576cf569b0e7bed6d7498f4e1a09

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/99086 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/18729 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/8973 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/104206 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/49667 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/101126 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/19018 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/27166 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/75418 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/32539 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/102091 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/14456 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/89473 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/55782 "Found 1 new API test failure: /WPE/TestCookieManager:/webkit/WebKitCookieManager/persistent-storage (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/14248 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/7445 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/49043 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/84177 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/7533 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/106569 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/26195 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/19090 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/84378 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/26558 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/85673 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/83888 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21290 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/28552 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/6226 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/19908 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/26136 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/31317 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/25957 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/29270 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/27523 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e1145058ea554c0b8b534acbf96a9c3f41a25b8c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/37242

Ahmad-S792 commented:
We already have it in `StyleAdjuster.cpp`, hence there is no progression. https://searchfox.org/wubkat/rev/596fac2b7c3264b575d234416a752f4386b8be56/Source/WebCore/style/StyleAdjuster.cpp#575

I am not sure which one is more performant or preferable so will leave it to @anttijk & @nt1m.

nt1m commented:
It's OK to do this but then we should remove the 2 lines in the StyleAdjuster

nt1m commented:
It's definitely more readable to have it in html.css.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ef4c2f1a4bce00335a49d6a64044f4fa08c0b1f5. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/37242

nt1m reviewed (approved):
Maybe refer to https://html.spec.whatwg.org/#the-marquee-element-2 in your commit message?

Ahmad-S792 commented:
@AlexsanderDamaceno - Can you also sync `html/obsolete/requirements-for-implementations/the-marquee-element-0/marquee-overflow.html`?

Ahmad-S792 commented:
@AlexsanderDamaceno - this would be quick turnaround - if you can rebase it and sync WPT tests? :-)

Ahmad-S792 commented:
> @AlexsanderDamaceno - Can you also sync `html/obsolete/requirements-for-implementations/the-marquee-element-0/marquee-overflow.html`?

Syncing tests here - https://github.com/WebKit/WebKit/pull/43671

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/86bffb52671f576cf569b0e7bed6d7498f4e1a09. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/37242

AlexsanderDamaceno committed:
Make the <marquee> element spec-compliant by adding overflow: hidden
https://bugs.webkit.org/show_bug.cgi?id=283811

Reviewed by Tim Nguyen.

Make the <marquee> element spec-compliant by adding  missing overflow: hidden property.

<marquee> element spec: https://html.spec.whatwg.org/#the-marquee-element-2

* Source/WebCore/css/html.css:
(marquee):
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjust const):

Canonical link: https://commits.webkit.org/293513@main

webkit-commit-queue commented:
Committed 293513@main (25d80dcb4ad3): <https://commits.webkit.org/293513@main>

Reviewed commits have been landed. Closing PR #37242 and removing active labels.
